event_model/__init__.py

```diff
@@ -155,6 +156,81 @@ def __delitem__(self, key):
             "The handler registry cannot be edited directly. "
             "Instead, use the method Filler.deregister_handler.")
 
+# A "coersion funcion" is a hook that Filler can use to, for example, ensure
+# all the external data read in my handlers is an *actual* numpy array as
+# opposed to some other array-like such as h5py.Dataset or dask.array.Array,
+# or wrap every result is dask.array.from_array(...).
+#
+# It has access to the handler_class as it is registered and to some state
+# provided by the Filler (more on that below). It is expected to return
+# something that is API-compatible with handler_class.  That might be
+# handler_class itself (a no-op), a subclass, or an altogether different class
+# with the same API. See example below.
+#
+# The "state provided by the Filler", mentioned above is passed into the
+# coersion functions below as ``filler_state``. It is a namespace containing
+# information that may be useful for the coersion functions.  Currently, it has
+# ``filler_state.descriptor`` and ``filler_state.key``. More may be added in
+# the future if the need arises. Ultimately, this is necessary because Resource
+# documents don't know the shape and dtype of the data that they reference.
+# That situation could be improved in the future; to some degree this is a
+# work-around.
+#
+# As an implementation detail, the ``filler_state`` is a ``threading.local``
+# object to ensure that filling is thread-safe.
+#
+# Third-party libraries can register custom coersion options via the
+# register_coersion function below. For example, databroker uses this to
+# register a 'delayed' option. This avoids introducing dependency on a specific
+# delayed-computation framework (e.g. dask) in event-model itself.
+
+
+def as_is(handler_class, filler_state):
+    "A no-op coersion function that returns handler_class unchanged."
+    return handler_class
+
+
+def force_numpy(handler_class, filler_state):
+    "A coersion that makes handler_class.__call__ return actual numpy.ndarray."
+    class Subclass(handler_class):
+        def __call__(*args, **kwargs):
+            raw_result = super().__call__(*args, **kwargs)
+            result_as_array = numpy.asarray(raw_result)
```

Comment:
```suggestion
            result_as_array = numpy.asanyarray(raw_result)
```

be a bit more forgiving on casting.